engine/core/src/main/java/org/datacleaner/metamodel/datahub/DatahubConnection.java

```diff
@@ -0,0 +1,176 @@
+/**
+ * DataCleaner (community edition)
+ * Copyright (C) 2014 Neopost - Customer Information Management
+ *
+ * This copyrighted material is made available to anyone wishing to use, modify,
+ * copy, or redistribute it subject to the terms and conditions of the GNU
+ * Lesser General Public License, as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this distribution; if not, write to:
+ * Free Software Foundation, Inc.
+ * 51 Franklin Street, Fifth Floor
+ * Boston, MA  02110-1301  USA
+ */
+package org.datacleaner.metamodel.datahub;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.http.HttpHost;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.impl.client.HttpClients;
+import org.datacleaner.util.SecurityUtils;
+import org.datacleaner.util.http.CASMonitorHttpClient;
+import org.datacleaner.util.http.HttpBasicMonitorHttpClient;
+import org.datacleaner.util.http.MonitorHttpClient;
+
+/**
+ * Describes the connection information needed to connect to the DataHub.
+ */
+
+public class DatahubConnection {
+
+    private final String _hostname;
+    private final int _port;
+    private final boolean _https;
+    private final String _tenantId;
+    private final String _username;
+    private final String _password;
+    private final String _contextPath = "/ui";
+    private final String _datahubContext = "/datastores/Golden%20record";
+    private final String _scheme;
+    // HttpClientContext _context;
+    private boolean _acceptUnverifiedSslPeers;
+    private final String _securityMode;
+
+    public DatahubConnection(String hostname, Integer port, String username,
+            String password, String tenantId, boolean https, boolean acceptUnverifiedSslPeers, String securityMode) {
+
+        _hostname = hostname;
+        _port = port;
+        _https = https;
+        _tenantId = tenantId;
+        _username = username;
+        _password = password;
+        _scheme = _https ? "https" : "http";
+        _acceptUnverifiedSslPeers = acceptUnverifiedSslPeers;
+        _securityMode = securityMode;
+
+        // CredentialsProvider credsProvider = new BasicCredentialsProvider();
+        // credsProvider.setCredentials(
+        // new AuthScope(AuthScope.ANY_HOST, AuthScope.ANY_PORT),
+        // new UsernamePasswordCredentials(getUsername(), getPassword()));
+        // _context = HttpClientContext.create();
+        // _context.setCredentialsProvider(credsProvider);
```

Comment:
Quite some commented code in this class. Is it needed anymore? If not let's erase it.